enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -271,7 +271,14 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
             Iterables.getFirst(findArtifactPaths(root, artifact), ImmutableList.of());
         if (DependencyGraphBuilder.requiredDependency(firstArtifactPath)) {
           logger.error("Could not find artifact " + artifact);
-          logger.error("Paths to the missing artifact: " + pathsToArtifact);
+          if (pathsToArtifact.isEmpty()) {
+            // Maven may throw ArtifactDescriptorException even when the artifact is not needed
```

Comment:
may throw or does throw?

Reply:
Updated to "On certain conditions, Maven throws ..." to clarify "may".